autopush-common/src/db/mod.rs

```diff
@@ -203,6 +207,27 @@ impl User {
     }
 }
 
+#[repr(u8)]
+#[derive(Debug, PartialEq, PartialOrd, Serialize, Deserialize, Clone, Copy)]
+#[serde(rename_all = "kebab-case")]
+pub enum Urgency {
+    VeryLow = 0,
+    Low = 1,
+    Normal = 2,
+    High = 3,
+}
+
+impl From<&str> for Urgency {
+    fn from(value: &str) -> Self {
+        match value.to_lowercase().as_str() {
+            "high" => Urgency::High,
+            "low" => Urgency::Low,
+            "very-low" => Urgency::VeryLow,
+            _ => Urgency::Normal,
+        }
+    }
+}
```

Comment:
Considering that the two times you're using a `from` here, you might want to consider adding a `impl From<Option<actix_web::http::HeaderValue>> for Urgency`

Reply:
I've added `From<Option<&String>>`